functions.php

```diff
@@ -125,6 +125,103 @@ function custom_login_logo() {
 
 // UTILITY FUNCTIONS
 
+// Return an array of exhibition types from these $posts.
+function get_exhibition_types($posts) {
+  if ( empty( $posts ) ) {
+    return '';
+  }
+
+  $terms = array();
+
+  foreach($posts->posts as $post) {
+
+    if ($post->post_type === 'exposiciones') {
+      $post_type = get_post_type_object( $post->post_type );
+      $post_terms = wp_get_post_terms( $post->ID, 'tipo_de_exposicion' );
+      foreach($post_terms as $post_term) {
+        echo in_array($post_term->slug, array_column($terms, 'slug'));
+        if ( in_array($post_term->slug, array_column($terms, 'slug')) == false ) {
+          $terms[] = array(
+            'slug'  =>  $post_term->slug,
+            'name'  =>  $post_term->name,
+          );
+        }
+      }
+    } 
+  }
+  //pr( $terms );
+  return $terms;
+}
+
+// Return an array of post types fropm these $posts
+function get_posts_taxonomies($posts) {
+  if ( empty( $posts ) ) {
+    return '';
+  }
+
+  $terms = array();
+
+  foreach($posts->posts as $post) {
+
+    if ($post->post_type !== 'exposiciones') {
+      $post_type = get_post_type_object( $post->post_type );
+      if ( in_array($post_type->name, array_column($terms, 'slug')) == false ) {
+        $terms[] = array(
+          'slug'  =>  $post_type->name,
+          'name'  =>  $post_type->label,
+        );
+      }
+    }
+
+  }
+  //pr( $terms );
+  return $terms;
```

Comment:
this function looks very similar to the previous. Perhaps better to make 1 function with another variable?